### PR TITLE
[PodTargetInstaller] De-duplicate files

### PR DESCRIPTION
**Pull request by segiddins**
Followup to https://github.com/CocoaPods/CocoaPods/pull/3553.

\c @orta @neonichu 

I can't get tests for this to work, sadly.

### CODE REVIEW

File: lib/cocoapods/installer/target_installer/pod_target_installer.rb [hunk 1]
```diff
@@ -57,17 +57,19 @@ def add_files_to_build_phases
...
+          file_added = -> (file_ref, phase_name) { native_target.send(phase_name).include?(file_ref) }
+
           {
             true => file_accessor.arc_source_files,
             false => file_accessor.non_arc_source_files,
           }.each do |arc, files|
             files = files - headers - other_source_files
             flags = compiler_flags_for_consumer(consumer, arc)
-            regular_file_refs = files.map { |sf| project.reference_for_path(sf) }
+            regular_file_refs = files.map { |sf| project.reference_for_path(sf) }.reject { |fr| p fr; file_added[fr, :source_build_phase] }
```
Comment (mrackwitz) [inline lib/cocoapods/installer/target_installer/pod_target_installer.rb]: Don't debug print the file reference here.

**Comment by orta** [comment]
:+1:

**Comment by mrackwitz** [comment]
Why shouldn't Xcodeproj `AbstractTarget#add_file_references` take care of that? It is implemented in a similar way already for it's counterpart in `AbstractBuildPhase`.

**Comment by segiddins** [comment]
¯\_(ツ)_/¯ I just basically copy-pasted orta's implementation. Feel free to make the change in Xcodeproj if you think its the proper place to do so.

**Pull request closed without merging**